Objective-C/CBLConnectedClient.mm

```diff
@@ -0,0 +1,214 @@
+//
+//  CBLConnectedClient.mm
+//  CouchbaseLite
+//
+//  Copyright (c) 2022 Couchbase, Inc All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+#import "CBLConnectedClient.h"
+#import "CBLStringBytes.h"
+#import "CBLWebSocket.h"
+#import "CBLCoreBridge.h"
+#import "CBLDocument+Internal.h"
+#import "CBLStatus.h"
+
+using namespace fleece;
+
+@interface CBLConnectedClient ()
+
+@property (readonly, nonatomic) dispatch_queue_t dispatchQueue;
+@property (nonatomic, weak) void(^getDocCompletion)(CBLDocument* docInfo, NSError* error);
+@property (nonatomic, weak) void(^updateDocCompletion)(BOOL success, NSError* error);
+
+@end
+
+@implementation CBLConnectedClient {
+    C4ConnectedClient*  _client;
+    NSURL*              _url;
+    C4Error             _c4err;
+}
+
+@synthesize dispatchQueue=_dispatchQueue;
+@synthesize getDocCompletion=_getDocCompletion;
+@synthesize updateDocCompletion=_updateDocCompletion;
+
+- (instancetype) initWithURL: (NSURL*)url authenticator: (CBLAuthenticator*)authenticator {
+    self = [super init];
+    
+    if (self) {
+        _url = url;
+        
+        NSString* qName = $sprintf(@"ConnectedClient <%@>", url.absoluteString);
+        _dispatchQueue = dispatch_queue_create(qName.UTF8String, DISPATCH_QUEUE_SERIAL);
+        
+        CBLStringBytes sliceURL(_url.absoluteString);
+        C4SocketFactory socketFactory = CBLWebSocket.socketFactory;
+        socketFactory.context = (__bridge void*)self;
+        
+        C4ConnectedClientParameters params = {
+            .url                = sliceURL,
+            .socketFactory      = &socketFactory,
+            .callbackContext    = (__bridge void*)self,
+        };
+        C4Error c4err = {};
+        _client = c4client_new(&params, &c4err);
+    }
+    
+    return self;
+}
+
+- (void) dealloc {
+    c4client_free(_client);
+    
+    _client = nil;
+}
+
+#pragma mark - Start & Stop
+
+// Note: Currently we are not using and exposing this method.
+// When connectedClient is created, it will automatically gets created.
+// Also in going forward, we will try to avoid exposing start() and stop().
+- (void) start {
```

Comment:
currently this is not getting exposed or called. Currently LiteCore is calling the start when it creates a new  connection. I will remove this for now, to avoid confusion.